libknet/handle.c

```diff
@@ -791,7 +791,7 @@ int knet_handle_free(knet_handle_t knet_h)
 	_close_epolls(knet_h);
 	_destroy_buffers(knet_h);
 	_close_socks(knet_h);
-	crypto_fini(knet_h);
+	crypto_fini(knet_h, KNET_MAX_CRYPTO_INSTANCES + 1);
```

Comment:
I think a comment here saying that the "+ 1" forces crypto_fini() to release all might be helpful here, otherwise it looks odd. Either that or an internal #define to make it clearer what's going on. I did a bit of a double-take when I saw the code that runs this so I don't think it's an obvious usage.